pkg/machine/qemu/machine.go

```diff
@@ -507,29 +516,87 @@ func (v *MachineVM) Start(name string, opts machine.StartOptions) error {
 		}
 	}
 
-	// If the qemusocketpath exists and the vm is off/down, we should rm
-	// it before the dial as to avoid a segv
-	if err := v.QMPMonitor.Address.Delete(); err != nil {
+	vlanSocket, err := machineSocket(v.Name, "vlan", "")
+	if err != nil {
 		return err
 	}
-	for i := 0; i < 6; i++ {
-		qemuSocketConn, err = net.Dial("unix", v.QMPMonitor.Address.GetPath())
-		if err == nil {
-			break
+
+	isFdVlanVM := false
+	cmdLine := v.CmdLine
+	for _, c := range v.CmdLine {
+		if c == fdVlanNetdev() {
+			isFdVlanVM = true
 		}
-		time.Sleep(wait)
-		wait++
 	}
-	if err != nil {
-		return err
+
+	cmdLine = propagateHostEnv(cmdLine)
+
+	// Show migration recommendation if machine was created with different settings
+	if isFdVlanVM != useFdVLan() {
+		logrus.Warn("Stored Podman Machine configuration doesn't match current settings")
+		if isFdVlanVM {
+			logrus.Warnf("Consider replacing %q with %q in machine config", fdVlanNetdev(), socketVlanNetdev(vlanSocket.GetPath()))
+		} else {
+			logrus.Warnf("Consider replacing %q with %q in machine config", socketVlanNetdev(vlanSocket.GetPath()), fdVlanNetdev())
+		}
 	}
-	defer qemuSocketConn.Close()
 
-	fd, err := qemuSocketConn.(*net.UnixConn).File()
+	// Disable graphic window when not in debug mode
+	// Done in start, so we're not suck with the debug level we used on init
+	if !logrus.IsLevelEnabled(logrus.DebugLevel) {
+		cmdLine = append(cmdLine, "-display", "none")
+	}
+
+	forwardSock, forwardState, forwarderProcess, err := v.startHostNetworking(vlanSocket)
 	if err != nil {
-		return err
+		return fmt.Errorf("unable to start host networking: %q", err)
+	}
+	defer func() {
+		_ = vlanSocket.Delete()
+	}()
+
+	if isFdVlanVM {
+		// If the qemusocketpath exists and the vm is off/down, we should rm
+		// it before the dial as to avoid a segv
+		if err := vlanSocket.Delete(); err != nil {
```

Comment:
I believe it is not a newly introduced race as it comes from the original code. I don't see how I can fix it, as this part is still pretty magical to me (was even more magical, when it reused QMP socket, still have memories from the days I started experimenting with this code). If in future this code will be phased out by Unix Socket VLAN, this would solve the race as well.